kin-devplatform-sdk/src/main/java/kin/devplatform/accountmanager/AccountManagerImpl.java

```diff
@@ -237,7 +245,59 @@ public void onError(Exception e) {
 	public void switchAccount(final int accountIndex, @NonNull final KinCallback<Boolean> callback,
 		@NonNull final IMigrationManagerCallbacks migrationManagerCallbacks) {
 		Logger.log(new Log().withTag(TAG).put("switchAccount", "start"));
-		startMigration(accountIndex, callback, migrationManagerCallbacks);
+		RestorableWalletApi restorableWalletApi = new RestorableWalletApi();
+		try {
+			handleRestorable(accountIndex, callback, migrationManagerCallbacks, restorableWalletApi);
+		} catch (ApiException e) {
+			Logger.log(new Log().priority(Log.ERROR).withTag(TAG).text("getIsRestorableWallet: error is " + e));
+			callback.onFailure(ErrorUtil.createWWalletWasNotCreatedInThisAppException());
+		}
+	}
+
+	//TODO maybe need a better method name...
+	private void handleRestorable(final int accountIndex, @NonNull final KinCallback<Boolean> callback,
+		@NonNull final IMigrationManagerCallbacks migrationManagerCallbacks,
+		RestorableWalletApi restorableWalletApi) throws ApiException {
+		restorableWalletApi.getIsRestorableWallet(blockchainSource.getPublicAddress(accountIndex),
+			new ApiCallback<RestorableWalletRequest>() {
+				@Override
+				public void onFailure(ApiException e, int statusCode,
+					Map<String, List<String>> responseHeaders) {
+					handler.post(new Runnable() {
+						@Override
+						public void run() {
+							Logger.log(
+								new Log().priority(Log.ERROR).withTag(TAG).text("getIsRestorableWallet: onFailure"));
+							callback.onFailure(ErrorUtil.createWWalletWasNotCreatedInThisAppException());
```

Comment:
we should pass the `ApiExcpetion` we got, this is not a case of "wallet was not created in this app"

Reply:
It is not the case but what should we show in the progress dialog in this case?
And to whom we should pass it? because it is the market place?
Or do you mean to pass it to caller(SettingsPresenter)?

Reply:
pass it to the caller (SettingsPresenter) and have the general error text o.w. it will be misleading to the user (he can retry here for instance)

Reply:
So what is the general error  text?